### PR TITLE
Issue 13280 - `this.sizeof` rejected as static array length in some cases

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=13280

### CODE REVIEW

File: src/declaration.c [hunk 1]
```diff
@@ -375,7 +375,7 @@ void AliasDeclaration::semantic(Scope *sc)
     if (s && ((s->getType() && type->equals(s->getType())) || s->isEnumMember()))
         goto L2;                        // it's a symbolic alias
 
-    type = type->addStorageClass(storage_class);
+    type = type->addSTC(storage_class);
```
Comment (yebblies) [inline src/declaration.c]: That's fairly disturbing.  Why do they look the same but do different things?

Comment (9rnsr) [inline src/declaration.c]: In here, type is not yet semantic analysed. Using `Type::addStrageClass` is valid for the already analyzed type. On the other hand, `Type::addSTC` is for the yet not analyzed type (therefore it's normally used in parsing stage, and rarely in template instantiation process).

Comment (yebblies) [inline src/declaration.c]: Interesting.  Should `Type::addSTC` be changed to assert that deco isn't set?

Comment (yebblies) [inline src/declaration.c]: Or the other one, or both maybe.

**Comment by yebblies** [comment]
Auto-merge toggled on

File: src/declaration.c (hunk 1 above)
Comment (9rnsr) [inline src/declaration.c]: Not possible. Built-in basic types (e.g. `int`) has already deco, so parsing const(int)` will run `Type::int32->addSTC(STCconst)`.

Comment (9rnsr) [inline src/declaration.c]: As a quick check, adding `assert(deco)` in `Type::addStorageClass` will break druntime build.

**Pull request merged**